### ISSUE TITLE
clean up repo

### ISSUE THREAD

**Issue by Remi-Gau** [nilearn/nilearn.github.io#4]
not very urgent but this repo is getting kind of big and this may come back to bite us if we need to do any operation on it at some point

suggestion : squash commits in between release and purge the history

**Comment by Remi-Gau** [comment]
@ymzayek @bthirion

**Comment by ymzayek** [comment]
Agreed

**Comment by bthirion** [comment]
Sorry, but what is the risk ?

**Comment by Remi-Gau** [comment]
not very high except that some operations that should be fast (clone, fetch) are very slow on the repo already and they may get to a point where they are so slow they prevent us from working on it

**Comment by AlexandreAbraham** [comment]
Whether you clean it or not, do not hesitate to use partial clone. Blobless and treeless clones can save you juge amounts of time. Basically, these clones download part of the history but not all the diffs. Those diffs are accessed only when needed.

**Comment by Remi-Gau** [comment]
I agree but that's when you need to those diffs that things may get sluggish. So it is definitely just a once in a blue moon use case (only discovered it when wanting to check what a page of the doc looked like in previous versions) but for long term maintenance it may be better.

**Comment by Remi-Gau** [comment]
yeah cloning the repo on a slow connection is quite sluggish

> what is the risk ?

not being able to clone the repo sounds quite risky

**Comment by Remi-Gau** [comment]
Note that what I have done to lighten the weight a bit is to squash commits in this repo between releases to at least purge the history from all the dev builds.
